source-connector/src/test/kotlin/org/neo4j/connectors/kafka/source/Neo4jCdcSourceNodesIT.kt

```diff
@@ -341,6 +343,142 @@ abstract class Neo4jCdcSourceNodesIT {
+      session: Session
+  ) {
+    session
+        .run(
+            "CREATE (n:TestSource) SET n = ${'$'}props",
+            mapOf("props" to mapOf("name" to "Jane", "surname" to "Doe", "age" to 42)))
+        .consume()
+    session
+        .run(
+            "MATCH (ts:TestSource {name: 'Jane'}) SET ts += ${'$'}props",
+            mapOf(
+                "props" to
+                    mapOf(
+                        "surname" to "Smith",
+                        "age" to "42",
+                        "dob" to LocalDateTime.of(1982, 1, 1, 0, 0, 0, 0))))
+        .consume()
+    session
+        .run(
+            "MATCH (ts:TestSource {name: 'Jane'}) SET ts += ${'$'}props",
+            mapOf("props" to mapOf("age" to 42, "dob" to LocalDate.of(1982, 1, 1))))
+        .consume()
+
+    TopicVerifier.create<ChangeEvent, ChangeEvent>(consumer)
+        .assertMessageValue { value ->
+          assertThat(value)
+              .hasEventType(NODE)
+              .hasOperation(CREATE)
+              .labelledAs("TestSource")
+              .hasNoBeforeState()
+              .hasAfterStateProperties(mapOf("name" to "Jane", "surname" to "Doe", "age" to 42L))
+        }
+        .assertMessageValue { value ->
+          assertThat(value)
+              .hasEventType(NODE)
+              .hasOperation(UPDATE)
+              .labelledAs("TestSource")
+              .hasBeforeStateProperties(mapOf("name" to "Jane", "surname" to "Doe", "age" to 42L))
+              .hasAfterStateProperties(
+                  mapOf(
+                      "name" to "Jane",
+                      "surname" to "Smith",
+                      "age" to "42",
+                      "dob" to LocalDateTime.of(1982, 1, 1, 0, 0, 0, 0)))
+        }
+        .assertMessageValue { value ->
+          assertThat(value)
+              .hasEventType(NODE)
+              .hasOperation(UPDATE)
+              .labelledAs("TestSource")
+              .hasBeforeStateProperties(
+                  mapOf(
+                      "name" to "Jane",
+                      "surname" to "Smith",
+                      "age" to "42",
+                      "dob" to LocalDateTime.of(1982, 1, 1, 0, 0, 0, 0)))
+              .hasAfterStateProperties(
+                  mapOf(
+                      "name" to "Jane",
+                      "surname" to "Smith",
+                      "age" to 42L,
+                      "dob" to LocalDate.of(1982, 1, 1)))
+        }
+        .verifyWithin(Duration.ofSeconds(30))
+  }
+
+  @Neo4jSource(
+      startFrom = "EARLIEST",
+      strategy = CDC,
+      cdc =
+          CdcSource(
+              patternsIndexed = true,
+              topics =
+                  arrayOf(
+                      CdcSourceTopic(
+                          topic = "cdc", patterns = arrayOf(CdcSourceParam("(:TestSource)"))))))
+  @Test
+  fun `should read each operation to a single topic`(
```

Comment:
maybe rename this... the "read ... to" bit froze my brain for a couple of seconds

Reply:
I had the same experience 😂 

Renamed all to `publish ...` so that it's more clear.